crates/astria-sequencer/src/slinky/oracle/state_ext.rs

```diff
@@ -0,0 +1,289 @@
+                .parse::<CurrencyPair>()
+                .context("failed to parse storage key suffix as currency pair")?;
+            currency_pairs.push(currency_pair);
+        }
+        Ok(currency_pairs)
+    }
+
+    #[instrument(skip(self))]
+    async fn get_next_currency_pair_id(&self) -> Result<u64> {
+        let Some(bytes) = self
+            .get_raw(NEXT_CURRENCY_PAIR_ID_KEY)
+            .await
+            .context("failed reading next currency pair id from state")?
+        else {
+            return Ok(0);
+        };
+        let Id(next_currency_pair_id) =
+            Id::try_from_slice(&bytes).context("invalid next currency pair id bytes")?;
+        Ok(next_currency_pair_id)
+    }
+
+    #[instrument(skip(self))]
+    async fn get_price_for_currency_pair(
+        &self,
+        currency_pair: &CurrencyPair,
+    ) -> Result<Option<QuotePrice>> {
+        let bytes = self
+            .get_raw(&currency_pair_price_storage_key(currency_pair))
+            .await
+            .context("failed to get price for currency pair from state")?;
+        match bytes {
+            Some(bytes) => {
+                let price =
+                    serde_json::from_slice(&bytes).context("failed to deserialize price")?;
+                Ok(Some(price))
+            }
+            None => Ok(None),
+        }
+    }
+}
+
+impl<T: StateRead + ?Sized> StateReadExt for T {}
+
+#[async_trait]
+pub(crate) trait StateWriteExt: StateWrite {
+    #[instrument(skip(self))]
+    fn put_currency_pair_id(&mut self, currency_pair: &CurrencyPair, id: u64) -> Result<()> {
+        let bytes = borsh::to_vec(&Id(id)).context("failed to serialize currency pair id")?;
+        self.put_raw(currency_pair_to_id_storage_key(currency_pair), bytes);
+        Ok(())
+    }
+
+    #[instrument(skip(self))]
+    fn put_currency_pair(&mut self, id: u64, currency_pair: CurrencyPair) -> Result<()> {
+        let bytes =
+            serde_json::to_vec(&currency_pair).context("failed to serialize currency pair")?;
+        self.put_raw(id_to_currency_pair_storage_key(id), bytes);
+        Ok(())
+    }
+
+    #[instrument(skip(self))]
+    fn put_num_currency_pairs(&mut self, num_currency_pairs: u64) -> Result<()> {
+        let bytes = borsh::to_vec(&Count(num_currency_pairs))
+            .context("failed to serialize number of currency pairs")?;
+        self.put_raw(NUM_CURRENCY_PAIRS_KEY.to_string(), bytes);
+        Ok(())
+    }
+
+    #[instrument(skip(self))]
+    fn put_num_removed_currency_pairs(&mut self, num_removed_currency_pairs: u64) -> Result<()> {
+        let bytes = borsh::to_vec(&Count(num_removed_currency_pairs))
+            .context("failed to serialize number of removed currency pairs")?;
+        self.put_raw(NUM_REMOVED_CURRENCY_PAIRS_KEY.to_string(), bytes);
+        Ok(())
+    }
+
+    #[instrument(skip(self))]
+    fn put_currency_pair_state_and_price(
```

Comment:
I'm not too clear on why we have separate entries for the state (which contains the price) and the price itself.

Seems like we put an initial value for state during `OracleComponent::init_chain`, and that value is never changed.  However, we have `get_currency_pair_state` which is called by `OracleService::get_price` and `OracleService::get_prices`.  So those two methods will always report the same initial price, nonce and id.

Meanwhile, the price value is updated via `apply_prices_from_vote_extensions`, but that price is never read.

Reply:
yep, it looks like skip has updated this to just store the state only: https://github.com/skip-mev/connect/blob/main/x/oracle/keeper/keeper.go#L196 will update this respectively